### PR TITLE
Toggle between Participant and Household modes

### LINKED ISSUE #106: Menu Option for Toggling Between Household & Participant Modes
Add an option in the menu to switch between Household and Participant sides of the app. Do a check to make sure setting fields are filled in first; if not, show error message with link to open app settings, but opening app settings should still be password-protected. The exact wording of the option in the STEPS app will be provided by the STEPS WHO team.

Additional tasks:
- [x] User should not be able to exit settings without having filled settings for participants and households
- [x] Fix bug where switching between household and participant loses the previous page settings

### PR DESCRIPTION
**Pull request by ekigamba**
See more at #106 
Fixes #106

### CODE REVIEW

**Comment by ekigamba** [comment]
**Reason for Blocked Status:** Waiting for correct wording at #106

**Comment by ekigamba** [comment]
**No longer blocked!**

File: src/main/java/com/onaio/steps/orchestrators/flows/HouseholdFlow.java [hunk 1]
```diff
@@ -84,59 +83,73 @@ public Intent getIntent() {
...
+        String formIdValue = ((TextView) activity.findViewById(R.id.form_id_household)).getText().toString().trim();
+        String minAgeValue = ((TextView) activity.findViewById(R.id.min_age_household)).getText().toString().trim();
+        String maxAgeValue = ((TextView) activity.findViewById(R.id.max_age_household)).getText().toString().trim();
+
+        errorFields = validateHouseHoldSettings(surveyIdValue, deviceIdValue, formIdValue, minAgeValue, maxAgeValue);
         if (errorFields != null && !errorFields.isEmpty())
             throw new InvalidDataException(activity, getStringValue(R.string.action_settings), errorFields);
+    }
 
+    public List<String> validateHouseHoldSettings(String surveyId, String deviceId, String formId, String minAge, String maxAge) {
```
Comment (jasonrogena) [inline src/main/java/com/onaio/steps/orchestrators/flows/HouseholdFlow.java]: Would be nice if you made tests for this.

File: src/main/java/com/onaio/steps/orchestrators/flows/ParticipantFlow.java [hunk 2]
```diff
@@ -57,34 +56,37 @@ public boolean canHandle(FlowType flowType) {
...
+        String formIdValue = ((TextView) activity.findViewById(R.id.form_id_participant)).getText().toString().trim();
+        String minAgeValue = ((TextView) activity.findViewById(R.id.min_age_participant)).getText().toString().trim();
+        String maxAgeValue = ((TextView) activity.findViewById(R.id.max_age_participant)).getText().toString().trim();
+
+        errorFields = validateParticipantsSettings(deviceIdValue, formIdValue, minAgeValue, maxAgeValue);
         if (errorFields != null && !errorFields.isEmpty())
             throw new InvalidDataException(activity, getStringValue(R.string.action_settings), errorFields);
+    }
 
+    public List<String> validateParticipantsSettings(String deviceId, String formId, String minAge, String maxAge) {
```
Comment (jasonrogena) [inline src/main/java/com/onaio/steps/orchestrators/flows/ParticipantFlow.java]: Would be nice if you added tests for this.

**Pull request merged**